sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/v2Commands.scala

```diff
@@ -45,7 +45,7 @@ trait V2WriteCommand extends Command {
           case (inAttr, outAttr) =>
             // names and types must match, nullability must be compatible
             inAttr.name == outAttr.name &&
-              DataType.equalsIgnoreCompatibleNullability(outAttr.dataType, inAttr.dataType) &&
+              DataType.equalsIgnoreCompatibleNullability(inAttr.dataType, outAttr.dataType) &&
```

Comment:
Hi, @HeartSaVioR . This looks like the same with `branch-2.4` but the issue is reported at 3.0.0+. Could you confirm that this is 3.0.0-only issue or not?
```scala
  override lazy val resolved: Boolean = {
    table.resolved && query.resolved && query.output.size == table.output.size &&
        query.output.zip(table.output).forall {
          case (inAttr, outAttr) =>
            // names and types must match, nullability must be compatible
            inAttr.name == outAttr.name &&
                DataType.equalsIgnoreCompatibleNullability(outAttr.dataType, inAttr.dataType) &&
                (outAttr.nullable || !inAttr.nullable)
        }
  }
```

Reply:
Ah thanks for noticing. Nice finding. I found it from V2WriteCommand so thought it was added later. Will check the code path in branch-2.4 and test it.

Reply:
The usage of AppendData is reverted in https://github.com/apache/spark/commit/b6e4aca0be7f3b863c326063a3c02aa8a1c266a3 for branch-2.4 and shipped to Spark 2.4.0. So while the code in AppendData is broken, it's a dead code.

We seem to have three options: 1) revert remaining part of AppendData in branch-2.4 2) fix the code but leave it as dead 3) leave it as it is. What's our preference?

cc. @cloud-fan @HyukjinKwon

Reply:
I prefer (2) because there is a downstream using it in their fork.